src/coreclr/tools/Common/TypeSystem/Ecma/EcmaType.TypeEquivalence.cs

```diff
@@ -80,12 +80,12 @@ private TypeIdentifierData ComputeTypeIdentifierData()
                     if (this.HasCustomAttribute("System.Runtime.InteropServices", "ImportedFromTypeLibAttribute") || this.HasCustomAttribute("System.Runtime.InteropServices", "PrimaryInteropAssemblyAttribute"))
                     {
```

Comment:
These look wrong too:  it should be checking the assembly for those attributes, not the current type.

Reply:
yeah, ImportedFromTypeLibAttribute "Indicates that the types defined within an assembly were originally defined in a type library" and its only valid target is an assembly.

Reply:
Huh, sigh. Yeah, this needs to be fixed too. It will slightly reduce the scope of your perf fix, but is necessary to avoid marking stuff as type equivalent.